core/lib/spree/permission_sets/user_management.rb

```diff
@@ -2,7 +2,16 @@ module Spree
   module PermissionSets
     class UserManagement < PermissionSets::Base
       def activate!
-        can :manage, Spree.user_class
+        can [:admin, :display, :create, :update], Spree.user_class
+
+        # due to how cancancan filters by associations,
+        # we have to define this twice, once for `accessible_by`
+        can :update_email, Spree.user_class, spree_roles: { id: nil }
+        # and once for `can?`
+        can :update_email, Spree.user_class do |user|
+          user.spree_roles.none?
```

Comment:
Interesting, OK. What do you guys think is the best thing to do here?  I can't think of any other great solution that doesn't make assumptions like this about roles.  The only other idea that comes to mind is just eliminating `update_email` from this `UserManagement` permission set entirely and letting stores add stuff like this inside their apps if they need it.  (I imagine many stores will just use the super-admin role for all admins anyway).  Thoughts?